### ISSUE TITLE
No web driver yet available for build [15A263e] + crash on boot

### ISSUE THREAD

**Issue by sammcj** [goalque/automate-eGPU#6]
```bash
samm-mbp ~/git/automate-eGPU % sudo ./automate-eGPU.sh
*** automate-eGPU.sh v0.9.6 - (c) 2015 by Goalque ***
-------------------------------------------------------
Detected eGPU
 GM204 [GeForce GTX 970]
Current OS X
 10.11 15A263e
Previous OS X
 [not found]
Latest installed Nvidia web driver
 Version: 346.03.01b07
 Source: 3rd Party
 Install Date: 8/29/15, 2:29 PM

Checking IOPCITunnelCompatible keys...

Missing IOPCITunnelCompatible keys.
Mac board-id found.
Searching for matching driver...

No web driver yet available for build [15A263e].
This script can download and modify the older package [346.03.01b07] (y/n)?
y
No web driver found for OS X 10.11.
Web driver not found. Nvidia may have changed their web driver search service.
```

The existing "older" package it's detecting I downloaded and modified to install manually.
Unfortunately upon reboot if the eGPU is plugged in OSX KP's on boot, unplugged it boots fine...

Then I swapped the thunderbolt port on my MBP and started with the eGPU plugged in:

```
NVIDIA Chip Model:

  Chipset Model:	NVIDIA Chip Model
  Type:	GPU
  Bus:	PCIe
  PCIe Lane Width:	x8
  Vendor:	NVIDIA (0x10de)
  Device ID:	0x13c2
  Revision ID:	0x00a1
  ROM Revision:	preset 1.0.0
  gMux Version:	3.2.19 [3.2.8]
```

```
samm-mbp ~/git/automate-eGPU % sudo ioreg| grep -iE 'thunderbolt|nvidia' | pbcopy

    | | | |       |     +-o AppleThunderboltHAL  <class AppleThunderboltHAL, id 0x100000269, registered, matched, active, busy 0 (172 ms), retain 9>
    | | | |       |       +-o AppleThunderboltNHIType2  <class AppleThunderboltNHIType2, id 0x10000026c, registered, matched, active, busy 0 (172 ms), retain 10>
    | | | |       |         +-o IOThunderboltController  <class IOThunderboltController, id 0x10000026d, registered, matched, active, busy 0 (172 ms), retain 22>
    | | | |       |           +-o IOThunderboltLocalNode  <class IOThunderboltLocalNode, id 0x100000271, registered, matched, active, busy 0 (44 ms), retain 7>
    | | | |       |           | +-o AppleThunderboltIPService  <class AppleThunderboltIPService, id 0x100000382, registered, matched, active, busy 0 (2 ms), retain 9>
    | | | |       |           |   +-o AppleThunderboltIPPort  <class AppleThunderboltIPPort, id 0x100000383, registered, matched, active, busy 0 (2 ms), retain 7>
    | | | |       |           |   +-o AppleThunderboltIPPort  <class AppleThunderboltIPPort, id 0x100000386, registered, matched, active, busy 0 (1 ms), retain 7>
    | | | |       |           +-o IOThunderboltPort@5  <class IOThunderboltPort, id 0x100000272, registered, matched, active, busy 0 (127 ms), retain 8>
    | | | |       |             +-o IOThunderboltSwitchType2  <class IOThunderboltSwitchType2, id 0x100000273, registered, matched, active, busy 0 (106 ms), retain 24>
    | | | |       |               +-o IOThunderboltPort@1  <class IOThunderboltPort, id 0x100000278, registered, matched, active, busy 0 (32 ms), retain 8>
    | | | |       |               +-o IOThunderboltPort@2  <class IOThunderboltPort, id 0x100000279, registered, matched, active, busy 0 (32 ms), retain 7>
    | | | |       |               +-o IOThunderboltPort@3  <class IOThunderboltPort, id 0x10000027a, registered, matched, active, busy 0 (105 ms), retain 13>
    | | | |       |               | +-o IOThunderboltPort@3  <class IOThunderboltPort, id 0x1000005bf, registered, matched, active, busy 0 (72 ms), retain 12>
    | | | |       |               |   +-o IOThunderboltSwitchType2  <class IOThunderboltSwitchType2, id 0x1000005c0, registered, matched, active, busy 0 (10 ms), retain 19>
    | | | |       |               |     +-o IOThunderboltPort@1  <class IOThunderboltPort, id 0x1000005c2, registered, matched, active, busy 0 (9 ms), retain 7>
    | | | |       |               |     +-o IOThunderboltPort@2  <class IOThunderboltPort, id 0x1000005c3, registered, matched, active, busy 0 (8 ms), retain 7>
    | | | |       |               |     +-o IOThunderboltPort@4  <class IOThunderboltPort, id 0x1000005c4, registered, matched, active, busy 0 (9 ms), retain 7>
    | | | |       |               |     +-o IOThunderboltPort@6  <class IOThunderboltPort, id 0x1000005c5, registered, matched, active, busy 0 (10 ms), retain 13>
    | | | |       |               |     | +-o AppleThunderboltPCIUpAdapter  <class AppleThunderboltPCIUpAdapter, id 0x1000005cb, registered, matched, active, busy 0 (0 ms), retain 6>
    | | | |       |               |     +-o IOThunderboltPort@7  <class IOThunderboltPort, id 0x1000005c6, registered, matched, active, busy 0 (9 ms), retain 7>
    | | | |       |               |     | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x1000005c9, registered, matched, active, busy 0 (1 ms), retain 5>
    | | | |       |               |     +-o IOThunderboltPort@8  <class IOThunderboltPort, id 0x1000005c7, registered, matched, active, busy 0 (9 ms), retain 7>
    | | | |       |               |     | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x1000005ca, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               |     +-o IOThunderboltPort@A  <class IOThunderboltPort, id 0x1000005c8, registered, matched, active, busy 0 (10 ms), retain 7>
    | | | |       |               |       +-o AppleThunderboltDPOutAdapter  <class AppleThunderboltDPOutAdapter, id 0x1000005cc, registered, matched, active, busy 0 (0 ms), retain 6>
    | | | |       |               +-o IOThunderboltPort@4  <class IOThunderboltPort, id 0x10000027b, registered, matched, active, busy 0 (32 ms), retain 7>
    | | | |       |               +-o IOThunderboltPort@6  <class IOThunderboltPort, id 0x10000027c, registered, matched, active, busy 0 (34 ms), retain 14>
    | | | |       |               | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x100000287, registered, matched, active, busy 0 (0 ms), retain 6>
    | | | |       |               +-o IOThunderboltPort@7  <class IOThunderboltPort, id 0x10000027d, registered, matched, active, busy 0 (34 ms), retain 7>
    | | | |       |               | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x100000286, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               +-o IOThunderboltPort@8  <class IOThunderboltPort, id 0x10000027e, registered, matched, active, busy 0 (32 ms), retain 7>
    | | | |       |               | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x100000282, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               +-o IOThunderboltPort@9  <class IOThunderboltPort, id 0x10000027f, registered, matched, active, busy 0 (32 ms), retain 7>
    | | | |       |               | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x100000283, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               +-o IOThunderboltPort@B  <class IOThunderboltPort, id 0x100000280, registered, matched, active, busy 0 (33 ms), retain 7>
    | | | |       |               | +-o AppleThunderboltDPInAdapter  <class AppleThunderboltDPInAdapter, id 0x100000285, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               +-o IOThunderboltPort@C  <class IOThunderboltPort, id 0x100000281, registered, matched, active, busy 0 (33 ms), retain 7>
    | | | |       |                 +-o AppleThunderboltDPInAdapter  <class AppleThunderboltDPInAdapter, id 0x100000284, registered, matched, active, busy 0 (0 ms), retain 5>
```

```
Echo Express III TB2:

  Vendor Name:	Sonnet Technologies, Inc.
  Device Name:	Echo Express III TB2
  Vendor ID:	0x8
  Device ID:	0x9
  Device Revision:	0x1
  UID:	0x0008000900131D70
  Route String:	3
  Firmware Version:	21.1
  Port (Upstream):
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 10 Gb/s x2
  Current Link Width:	0x1
  Cable Firmware Version:	0.1.23
  Link Controller Firmware Version:	0.13.0
  Port:
  Status:	No device connected
  Link Status:	0x7
  Speed:	Up to 10 Gb/s x2
  Current Link Width:	0x1
  Link Controller Firmware Version:	0.13.0
```

But it doesn't do anything.

**Comment by sammcj** [comment]
```
    | | | |       |     +-o AppleThunderboltHAL  <class AppleThunderboltHAL, id 0x100000269, registered, matched, active, busy 0 (172 ms), retain 9>
    | | | |       |       +-o AppleThunderboltNHIType2  <class AppleThunderboltNHIType2, id 0x10000026c, registered, matched, active, busy 0 (172 ms), retain 10>
    | | | |       |         +-o IOThunderboltController  <class IOThunderboltController, id 0x10000026d, registered, matched, active, busy 0 (172 ms), retain 22>
    | | | |       |           +-o IOThunderboltLocalNode  <class IOThunderboltLocalNode, id 0x100000271, registered, matched, active, busy 0 (44 ms), retain 7>
    | | | |       |           | +-o AppleThunderboltIPService  <class AppleThunderboltIPService, id 0x100000382, registered, matched, active, busy 0 (2 ms), retain 9>
    | | | |       |           |   +-o AppleThunderboltIPPort  <class AppleThunderboltIPPort, id 0x100000383, registered, matched, active, busy 0 (2 ms), retain 7>
    | | | |       |           |   +-o AppleThunderboltIPPort  <class AppleThunderboltIPPort, id 0x100000386, registered, matched, active, busy 0 (1 ms), retain 7>
    | | | |       |           +-o IOThunderboltPort@5  <class IOThunderboltPort, id 0x100000272, registered, matched, active, busy 0 (127 ms), retain 8>
    | | | |       |             +-o IOThunderboltSwitchType2  <class IOThunderboltSwitchType2, id 0x100000273, registered, matched, active, busy 0 (106 ms), retain 24>
    | | | |       |               +-o IOThunderboltPort@1  <class IOThunderboltPort, id 0x100000278, registered, matched, active, busy 0 (32 ms), retain 8>
    | | | |       |               +-o IOThunderboltPort@2  <class IOThunderboltPort, id 0x100000279, registered, matched, active, busy 0 (32 ms), retain 7>
    | | | |       |               +-o IOThunderboltPort@3  <class IOThunderboltPort, id 0x10000027a, registered, matched, active, busy 0 (105 ms), retain 13>
    | | | |       |               | +-o IOThunderboltPort@3  <class IOThunderboltPort, id 0x1000005bf, registered, matched, active, busy 0 (72 ms), retain 12>
    | | | |       |               |   +-o IOThunderboltSwitchType2  <class IOThunderboltSwitchType2, id 0x1000005c0, registered, matched, active, busy 0 (10 ms), retain 19>
    | | | |       |               |     +-o IOThunderboltPort@1  <class IOThunderboltPort, id 0x1000005c2, registered, matched, active, busy 0 (9 ms), retain 7>
    | | | |       |               |     +-o IOThunderboltPort@2  <class IOThunderboltPort, id 0x1000005c3, registered, matched, active, busy 0 (8 ms), retain 7>
    | | | |       |               |     +-o IOThunderboltPort@4  <class IOThunderboltPort, id 0x1000005c4, registered, matched, active, busy 0 (9 ms), retain 7>
    | | | |       |               |     +-o IOThunderboltPort@6  <class IOThunderboltPort, id 0x1000005c5, registered, matched, active, busy 0 (10 ms), retain 13>
    | | | |       |               |     | +-o AppleThunderboltPCIUpAdapter  <class AppleThunderboltPCIUpAdapter, id 0x1000005cb, registered, matched, active, busy 0 (0 ms), retain 6>
    | | | |       |               |     +-o IOThunderboltPort@7  <class IOThunderboltPort, id 0x1000005c6, registered, matched, active, busy 0 (9 ms), retain 7>
    | | | |       |               |     | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x1000005c9, registered, matched, active, busy 0 (1 ms), retain 5>
    | | | |       |               |     +-o IOThunderboltPort@8  <class IOThunderboltPort, id 0x1000005c7, registered, matched, active, busy 0 (9 ms), retain 7>
    | | | |       |               |     | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x1000005ca, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               |     +-o IOThunderboltPort@A  <class IOThunderboltPort, id 0x1000005c8, registered, matched, active, busy 0 (10 ms), retain 7>
    | | | |       |               |       +-o AppleThunderboltDPOutAdapter  <class AppleThunderboltDPOutAdapter, id 0x1000005cc, registered, matched, active, busy 0 (0 ms), retain 6>
    | | | |       |               +-o IOThunderboltPort@4  <class IOThunderboltPort, id 0x10000027b, registered, matched, active, busy 0 (32 ms), retain 7>
    | | | |       |               +-o IOThunderboltPort@6  <class IOThunderboltPort, id 0x10000027c, registered, matched, active, busy 0 (34 ms), retain 14>
    | | | |       |               | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x100000287, registered, matched, active, busy 0 (0 ms), retain 6>
    | | | |       |               +-o IOThunderboltPort@7  <class IOThunderboltPort, id 0x10000027d, registered, matched, active, busy 0 (34 ms), retain 7>
    | | | |       |               | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x100000286, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               +-o IOThunderboltPort@8  <class IOThunderboltPort, id 0x10000027e, registered, matched, active, busy 0 (32 ms), retain 7>
    | | | |       |               | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x100000282, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               +-o IOThunderboltPort@9  <class IOThunderboltPort, id 0x10000027f, registered, matched, active, busy 0 (32 ms), retain 7>
    | | | |       |               | +-o AppleThunderboltPCIDownAdapter  <class AppleThunderboltPCIDownAdapter, id 0x100000283, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               +-o IOThunderboltPort@B  <class IOThunderboltPort, id 0x100000280, registered, matched, active, busy 0 (33 ms), retain 7>
    | | | |       |               | +-o AppleThunderboltDPInAdapter  <class AppleThunderboltDPInAdapter, id 0x100000285, registered, matched, active, busy 0 (0 ms), retain 5>
    | | | |       |               +-o IOThunderboltPort@C  <class IOThunderboltPort, id 0x100000281, registered, matched, active, busy 0 (33 ms), retain 7>
    | | | |       |                 +-o AppleThunderboltDPInAdapter  <class AppleThunderboltDPInAdapter, id 0x100000284, registered, matched, active, busy 0 (0 ms), retain 5>
```

**Comment by sammcj** [comment]
Interesting, I just rebooted about 3 times and it finally started with the eGPU plugged in:

```
NVIDIA Chip Model:

  Chipset Model:	NVIDIA Chip Model
  Type:	GPU
  Bus:	PCIe
  PCIe Lane Width:	x8
  Vendor:	NVIDIA (0x10de)
  Device ID:	0x13c2
  Revision ID:	0x00a1
  ROM Revision:	preset 1.0.0
  gMux Version:	3.2.19 [3.2.8]
```

```
Echo Express III TB2:

  Vendor Name:	Sonnet Technologies, Inc.
  Device Name:	Echo Express III TB2
  Vendor ID:	0x8
  Device ID:	0x9
  Device Revision:	0x1
  UID:	0x0008000900131D70
  Route String:	3
  Firmware Version:	21.1
  Port (Upstream):
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 10 Gb/s x2
  Current Link Width:	0x1
  Cable Firmware Version:	0.1.23
  Link Controller Firmware Version:	0.13.0
  Port:
  Status:	No device connected
  Link Status:	0x7
  Speed:	Up to 10 Gb/s x2
  Current Link Width:	0x1
  Link Controller Firmware Version:	0.13.0
```

**Comment by LeonimuZ** [comment]
Exact same boat! It was working perfectly on Yosemite.

**Comment by goalque** [comment]
Yes, this is a known issue and rare situation. You first downloaded the package 346.03.01b07 with [-url] option, right? The script assumes that all the packages are available through the web API or web page, so it suggests to download and modify it and of course cannot find, since it’s beta. Normally, users'  downloaded packages are not those leaked beta drivers from Nvidia :)

If you want to reinstall the pkg for 10.11, please use [-uninstall] option eGPU disconnected, then hot-plug and run with [-url] option. Does it work?

**Comment by LeonimuZ** [comment]
I'm on the same boat! Uninstalled the old drivers and then tried with the -url option and the script goes smoothly but the GPU still doesn't work. Hangs with eGPU plugged in, and works without it.

**Comment by goalque** [comment]
Do you happen to have a multi-slot enclosure? If so, try turning on automatic [-a] mode and restart. If you have edited kexts manually before, please reinstall OS X. I’ve been playing with AMDs recently, so I will double confirm if 10.11 (15A263e) still works with the Nvidia eGPU.

**Comment by LeonimuZ** [comment]
No, I don't have a multi-slot enclosure, but tried running it in Automatic and it's the same. I had not edited the kext files manually before, it was all done through the script. It worked perfect on Yosemite, so I know is not a hardware related issue. I didn't do a clean install of El Capitan, I updated (which should have replaced all of the Yosemite system files and therefore remove any modified files).

**Comment by goalque** [comment]
I just tested with the 2014 Mac mini and 2014 15" rMBP Iris Pro without issues (used the same Samsung T1 portable USB3 SSD). DP, HDMI and DVI interfaces all worked. What's your Mac model?
```
goalques-MacBook-Pro:Desktop goalque$ sudo ./automate-eGPU.sh -url http://us.download.nvidia.com/tYGrxreBnE8b89jZ0VysebeSwZJehZTa/WebDriver-346.03.01b07.pkg
Password:
*** automate-eGPU.sh v0.9.6 - (c) 2015 by Goalque ***
-------------------------------------------------------
Detected eGPU
 GM204 [GeForce GTX 980]
Current OS X
 10.11 15A263e
Previous OS X
 [not found]
Latest installed Nvidia web driver
 Version: 346.03.01b07
 Source: 3rd Party
 Install Date: 8/30/15, 11:26 PM

You are running official Nvidia driver.
Checking IOPCITunnelCompatible keys...

Missing IOPCITunnelCompatible keys.
Mac board-id not found.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 49.4M  100 49.4M    0     0  3209k      0  0:00:15  0:00:15 --:--:-- 4853k
Removing validation checks...
Modified package ready. Do you want to install (y/n)?
y
installer: Package name is NVIDIA Web Driver 346.03.01b07
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: The install requires restarting now.
Checking IOPCITunnelCompatible keys...

Missing IOPCITunnelCompatible keys.
IOPCITunnelCompatible mods done.
Board-id added.
All ready. Please restart the Mac.
goalques-MacBook-Pro:Desktop goalque$
```
When you type nvram -p, you should see the following line:
```
csr-active-config	w%00%00%00
```

**Comment by LeonimuZ** [comment]
To answer your previous questions: I have a Mid-2014 13" rMBP and I saw the *csr-active-config   w%00%00%00* even when it wasn't working.

But it works now all of the sudden!

I did the following:

1. Uninstalled the script and the Nvidia Driver then restarted
2. Copy pasted the run command from your test
3. Restarted and it worked all of the sudden.

I was using the exact same driver from the exact same link and it didn't work before.
Is there some sort of log I can get you? I would like for you to see what changed.
